sympy/geometry/polygon.py

```diff
@@ -402,6 +403,54 @@ def centroid(self):
             cy += v*(y1 + y2)
         return Point(simplify(A*cx), simplify(A*cy))
 
+    def second_moment_of_area(self,p=None):
+        """Second moment of area of polygon.
+
+        Notes
+        =====
+
+        about the axis passing through centroid if point is not given othervise about the axis passing through given point.
+
+        Parameters
+        ==========
+
+        p : Point, None
+            Default value is None.
+
+        Examples
+        ========
+
+        >>> from sympy import Point,Polygon,symbols
+        >>> a,b=symbols('a,b')
+        >>> p1,p2,p3,p4,p5=[(0,0),(a,0),(a,b),(0,b),(a/3,b/3)]
+        >>> rectangle=Polygon(p1,p2,p3,p4)
+        >>> rectangle.second_moment_of_area()
+        (a*b**3/12, a**3*b/12)
+        >>> rectangle.second_moment_of_area(p5)
+        (a*b**3/9, a**3*b/9)
+
+        """
+        I_xx,I_yy=0,0
+        args= self.args
+        for i in range(len(args)):
+            x1,y1 = args[i-1].args
+            x2,y2 = args[i].args
+            v = x1*y2 - x2*y1
+            I_xx += (y1**2+y1*y2+y2**2)*v
+            I_yy += (x1**2+x1*x2+x2**2)*v
+        A=self.area
+        c_x=self.centroid[0]
+        c_y=self.centroid[1]
+        #parallel axis theorem
+        I_xx_c = (I_xx/12) - A*(c_y**2)
+        I_yy_c = (I_yy/12) - A*(c_x**2)
+        if(p == None):
+            return I_xx_c,I_yy_c
+        I_xx = (I_xx_c + A*((p[1]-c_y)**2))
+        I_yy = (I_yy_c + A*((p[0]-c_x)**2))
+
+        return I_xx,I_yy
```

Comment:
There is still no product of area term.

Reply:
can be added using the fundamental formula of the product of area using integration I_xy=integrate(x*y,dA).but it will be slow as it uses integration.I am trying to get expression similar to  I_xx and I_yy.

Reply:
Can you compute the integral with SymPy, simplify it and then hard code it here?

Reply:
#for quadrilateral
p1,p2,p3,p4=Point(x1,y1),Point(x2,y2),Point(x3,y3),Point(x4,y4)

eq1_y=(((p1.y-p2.y)/(p1.x-p2.x))*(x-p1.x))+p1.y
eq2_y=(((p2.y-p3.y)/(p2.x-p3.x))*(x-p2.x))+p2.y
eq3_y=(((p3.y-p4.y)/(p3.x-p4.x))*(x-p3.x))+p3.y
eq4_y=(((p4.y-p1.y)/(p4.x-p1.x))*(x-p4.x))+p4.y

i1=integrate((eq1_y**2)*x,(x,x1,x2))
i2=integrate((eq2_y**2)*x,(x,x2,x3))
i3=integrate((eq3_y**2)*x,(x,x3,x4))
i4=integrate((eq4_y**2)*x,(x,x1,x4))
i_xy=simplify(i4-i1-i2-i3)

Reply:
#for quadrilateral
p1,p2,p3,p4=Point(x1,y1),Point(x2,y2),Point(x3,y3),Point(x4,y4)

eq1_y=(((p1.y-p2.y)/(p1.x-p2.x))(x-p1.x))+p1.y
eq2_y=(((p2.y-p3.y)/(p2.x-p3.x))(x-p2.x))+p2.y
eq3_y=(((p3.y-p4.y)/(p3.x-p4.x))(x-p3.x))+p3.y
eq4_y=(((p4.y-p1.y)/(p4.x-p1.x))(x-p4.x))+p4.y

i1=integrate((eq1_y**2)*x,(x,x1,x2))
i2=integrate((eq2_y**2)*x,(x,x2,x3))
i3=integrate((eq3_y**2)*x,(x,x3,x4))
i4=integrate((eq4_y**2)*x,(x,x1,x4))
i_xy=simplify(i4-i1-i2-i3)

Reply:
#for quadrilateral
p1,p2,p3,p4=Point(x1,y1),Point(x2,y2),Point(x3,y3),Point(x4,y4)

eq1_y=(((p1.y-p2.y)/(p1.x-p2.x))(x-p1.x))+p1.y
eq2_y=(((p2.y-p3.y)/(p2.x-p3.x))(x-p2.x))+p2.y
eq3_y=(((p3.y-p4.y)/(p3.x-p4.x))(x-p3.x))+p3.y
eq4_y=(((p4.y-p1.y)/(p4.x-p1.x))(x-p4.x))+p4.y

i1=integrate((eq1_y^2)*x,(x,x1,x2))
i2=integrate((eq2_y^2)*x,(x,x2,x3))
i3=integrate((eq3_y^2)*x,(x,x3,x4))
i4=integrate((eq4_y^2)*x,(x,x1,x4))
i_xy=simplify(i4-i1-i2-i3)

Reply:
Can you hard code the I_xy result you show above?

Reply:
I can hard code it but I think I can get some expression for I_xy. I have got some particular common term in the expression.
sir for the assertion part should I get the expression for I_xx and I_yy from fundamental formula and equate it with the implemented function for simple shape(like rectangle right angle triangle and regular hexagon and pentagon)

Reply:
Done